### PR TITLE
Hotfix/docker

### PR DESCRIPTION
**Pull request by distributedstatemachine**
## Description

This PR

- removes references to `--ws-*` from the docker compose file
- changes the repo from docker to this repo in the compose file 
- adds a ci to build an image when a new tag is published , and commits to main.

## Related Issue(s)

- Closes #[issue number]

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

## Breaking Change

If this PR introduces a breaking change, please provide a detailed description of the impact and the migration path for existing applications.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please include any relevant screenshots or GIFs that demonstrate the changes made.

## Additional Notes

Please provide any additional information or context that may be helpful for reviewers.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
```
Comment (sam0x17) [inline .github/workflows/docker.yml]:
```suggestion
    runs-on: SubtensorCI
```

**Review by sam0x17 — DISMISSED** [review]
one nit

File: .github/workflows/docker.yml (hunk 1 above)
Comment (distributedstatemachine) [reply .github/workflows/docker.yml]: Yeh makes sense. Do we have docker installed on it though ?

**Review by gztensor — APPROVED** [review]
It works locally

**Review by JohnReedV — APPROVED** [review]

**Review by open-junius — APPROVED** [review]
Looks good to me!

**Pull request merged**